### PR TITLE
added request body/data matching

### LINKED ISSUE #10: Match request body for PUT and POST requests
Hello! 

I started to use this library since I switched my work from Angular to React and axios for http-requests. Thank you for this package, that was really a missing feature.

Now I have a lack of matching body for requests with payload. I'd like to do something like this:

```js
mock.onPost('/api/users', {name: 'new user', email: 'test@email.com'}).reply(200);
```

Then I can to test not only that request was actually called, but also that it has proper body. That feature was built-in in Angular's mock library, so it would be nice to have it there as well.

### PR DESCRIPTION
**Pull request by wirmar**
fixes #10 
I used JSON.stringify for comparison, since the request body/data in the axios request config is a JSON string anyway, so no dependency added.
Since I didn't add filtering to only allow post/put/patch for this other types (get etc) will not match a request handler with a body.

All tests and linting are green on my machine.

### CODE REVIEW

**Comment by ctimmerm** [comment]
Great, thanks a lot for the pull request! If I'm not mistaken, `JSON.stringify` makes no guarantees about the order of the keys of an object (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify), so it's probably safer to check for deep equality anyway?

**Comment by wirmar** [comment]
Okay I did not know that, good point. I will adjust the pull request to use deep equality then.

File: src/utils.js [hunk 1]
```diff
@@ -8,16 +9,29 @@ function find(array, predicate) {
...
+      return handler[0].test(url) && isBodyMatching(body, handler[4]);
     }
   });
 }
 
+function isBodyMatching(body, requiredBody) {
+  var parsedBody;
+  var isJSON = true;
+  try {
+    parsedBody = JSON.parse(body);
```
Comment (wirmar) [inline src/utils.js]: This code here is a bit ugly, but JSON.parse with a non json argument throws an error and the request body could also be a simple string for example..

Comment (just-boris) [inline src/utils.js]: Actually we can just serialize `requiredBody` and compare them as strings.

Comment (just-boris) [inline src/utils.js]: You can just rely on parsedBody variable

```js
function isBodyMatching(body, requiredBody) {
    if(requiredBody === undefined) {
       return true;
    }
    var parsedBody
    try {
        parsedBody = JSON.parse(body);
    }
    catch (e) {}

    return parsedBody ? eql(parsedBody, requiredBody) : eql(body, requiredBody)
}
```

**Pull request merged**